src/xsettings.c

```diff
@@ -974,24 +974,6 @@ xsettings_get_system_font (void)
   return current_mono_font;
 }
 
-DEFUN ("tool-bar-get-system-style", Ftool_bar_get_system_style,
-       Stool_bar_get_system_style, 0, 0, 0,
-       doc: /* Get the system tool bar style.
-If no system tool bar style is known, return `tool-bar-style' if set to a
-known style.  Otherwise return image.  */)
-  (void)
-{
-  if (EQ (Vtool_bar_style, Qimage)
-      || EQ (Vtool_bar_style, Qtext)
-      || EQ (Vtool_bar_style, Qboth)
-      || EQ (Vtool_bar_style, Qboth_horiz)
-      || EQ (Vtool_bar_style, Qtext_image_horiz))
-    return Vtool_bar_style;
-  if (!NILP (current_tool_bar_style))
```

Comment:
I did not look at this closely enough last time since I was focused on the `match`. `current_tool_bar_style` is only returned when it has a value. The nil check in the `match` is incorrect. The port should read:

```
match style {
      Qimage | Qtext | Qboth | Qboth_horiz | Qtext_image_horiz => style,
      _ if  unsafe { current_tool_bar_style.is_nil() } => Qimage,
      _ => unsafe { current_tool_bar_style },
}
```

Reply:
Good catch! I pushed a fixed version now. I used another local variable to fetch the value of `current_tool_bar_style` to keep the number of unsafe blocks down.